velox/dwio/parquet/tests/ParquetTpchTestBase.h

```diff
@@ -127,53 +166,95 @@ class ParquetTpchTestBase : public testing::Test {
-          R"(COPY (SELECT l_orderkey, l_partkey, l_suppkey, l_linenumber,
-         l_quantity::DOUBLE as quantity, l_extendedprice::DOUBLE as extendedprice, l_discount::DOUBLE as discount,
-         l_tax::DOUBLE as tax, l_returnflag, l_linestatus, l_shipdate AS shipdate, l_commitdate, l_receiptdate,
-         l_shipinstruct, l_shipmode, l_comment FROM {})
-         TO '{}'(FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "orders",
-          R"(COPY (SELECT o_orderkey, o_custkey, o_orderstatus,
-         o_totalprice::DOUBLE as o_totalprice,
-         o_orderdate, o_orderpriority, o_clerk, o_shippriority, o_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "customer",
-          R"(COPY (SELECT c_custkey, c_name, c_address, c_nationkey, c_phone,
-         c_acctbal::DOUBLE as c_acctbal, c_mktsegment, c_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "nation",
-          R"(COPY (SELECT * FROM {})
-          TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "region",
-          R"(COPY (SELECT * FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "part",
-          R"(COPY (SELECT p_partkey, p_name, p_mfgr, p_brand, p_type, p_size,
-         p_container, p_retailprice::DOUBLE, p_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "supplier",
-          R"(COPY (SELECT s_suppkey, s_name, s_address, s_nationkey, s_phone,
-         s_acctbal::DOUBLE, s_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
-      std::make_pair(
-          "partsupp",
-          R"(COPY (SELECT ps_partkey, ps_suppkey, ps_availqty,
-         ps_supplycost::DOUBLE as supplycost, ps_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))")};
+  static std::shared_ptr<DuckDbQueryRunner> duckDb_;
+  static std::shared_ptr<TempDirectoryPath> tempDirectory_;
+  static TpchQueryBuilder tpchBuilder_;
+  static std::unordered_map<std::string, const std::string>
+      kDuckDbParquetWriteSQL_;
 };
 
+std::shared_ptr<DuckDbQueryRunner> ParquetTpchTestBase::duckDb_ = nullptr;
+std::shared_ptr<TempDirectoryPath> ParquetTpchTestBase::tempDirectory_ =
+    nullptr;
+TpchQueryBuilder ParquetTpchTestBase::tpchBuilder_ =
+    TpchQueryBuilder(dwio::common::FileFormat::PARQUET);
+
+std::unordered_map<std::string, const std::string>
+    ParquetTpchTestBase::kDuckDbParquetWriteSQL_ = {
+        std::make_pair(
+            "lineitem",
+            R"(SELECT l_orderkey,
+                      l_partkey,
+                      l_suppkey,
+                      l_linenumber,
+                      L_QUANTITY::DOUBLE AS quantity,
```

Comment:
Why cast to Double?

Reply:
Thanks for clarifying. Can we remove the cast to double? We now support decimal types.

Reply:
Got it, I've removed all the `cast to double` and rerun the tpch test, all passed.